go/pools/resource_pool_test.go

```diff
@@ -123,7 +123,7 @@ func TestOpen(t *testing.T) {
 	}
 	r.Close()
 	p.Put(nil)
-	if count.Get() != 4 {
+	if count.Get() != 5 {
```

Comment:
The error message below doesn't match up with this expectation... 
Also we should clarify with a comment that calling `Put` reopens the resource